cmd/vc-rest/startcmd/start.go

```diff
@@ -408,36 +468,60 @@ func acceptsDID(method string) bool {
-	var provider storage.Provider
+type edgeServiceProviders struct {
+	provider           storage.Provider
+	kmsSecretsProvider ariesstorage.Provider
+}
+
+func createStoreProviders(parameters *vcRestParameters) (edgeServiceProviders, error) {
+	var edgeServiceProvs edgeServiceProviders
 
-	var ariesProvider ariesstorage.Provider
+	checkForSameDBParams(&parameters.dbParameters)
 
 	switch {
-	case strings.EqualFold(parameters.databaseType, databaseTypeMemOption):
-		provider = memstore.NewProvider()
-		ariesProvider = ariesmemstorage.NewProvider()
-	case strings.EqualFold(parameters.databaseType, databaseTypeCouchDBOption):
+	case strings.EqualFold(parameters.dbParameters.databaseType, databaseTypeMemOption):
+		edgeServiceProvs.provider = memstore.NewProvider()
+	case strings.EqualFold(parameters.dbParameters.databaseType, databaseTypeCouchDBOption):
 		var err error
-		provider, err = couchdbstore.NewProvider(parameters.databaseURL,
-			couchdbstore.WithDBPrefix(parameters.databasePrefix))
 
+		edgeServiceProvs.provider, err = couchdbstore.NewProvider(parameters.dbParameters.databaseURL,
+			couchdbstore.WithDBPrefix(parameters.dbParameters.databasePrefix))
 		if err != nil {
-			return nil, nil, err
+			return edgeServiceProviders{}, err
 		}
+	default:
+		return edgeServiceProviders{}, fmt.Errorf("database type not set to a valid type." +
+			" run start --help to see the available options")
+	}
+
+	switch {
+	case strings.EqualFold(parameters.dbParameters.kmsSecretsDatabaseType, databaseTypeMemOption):
+		edgeServiceProvs.kmsSecretsProvider = ariesmemstorage.NewProvider()
+	case strings.EqualFold(parameters.dbParameters.kmsSecretsDatabaseType, databaseTypeCouchDBOption):
+		var err error
 
-		ariesProvider, err = ariescouchdbstorage.NewProvider(parameters.databaseURL,
-			ariescouchdbstorage.WithDBPrefix(parameters.databasePrefix))
+		edgeServiceProvs.kmsSecretsProvider, err =
+			ariescouchdbstorage.NewProvider(parameters.dbParameters.kmsSecretsDatabaseURL,
+				ariescouchdbstorage.WithDBPrefix(parameters.dbParameters.kmsSecretsDatabasePrefix))
 		if err != nil {
-			return nil, nil, err
+			return edgeServiceProviders{}, err
 		}
 
 	default:
-		return nil, nil, fmt.Errorf("database type not set to a valid type." +
+		return edgeServiceProviders{}, fmt.Errorf("key database type not set to a valid type." +
 			" run start --help to see the available options")
 	}
 
-	return provider, ariesProvider, nil
+	return edgeServiceProvs, nil
+}
+
+func checkForSameDBParams(dbParams *dbParameters) {
+	if strings.EqualFold(dbParams.databaseType, dbParams.kmsSecretsDatabaseType) &&
+		strings.EqualFold(dbParams.databaseURL, dbParams.kmsSecretsDatabaseURL) {
+		log.Warnln("Database and KMS secrets database both set to the same provider. It's recommended to use a " +
+			"separate provider for storage of KMS secrets to ensure that the provider hosting the EDVs " +
+			"cannot read the stored encrypted documents.")
```

Comment:
This is the warning I show in the log when the user sets both databases to the same provider. @troyronda @fqutishat @sandrask

Reply:
looks good